### PR TITLE
So 53 refactoring

### PR DESCRIPTION
**Pull request by jvniorrr**
- Fixed application folder layout to follow our diagram
- Tweaked some code for responsiveness, and few minor things.
- Cherry picked Miguels code for error and signup popups for Signup page
- Sign up page responsiveness fixed
- Side Navbar functionality implemented
- Top Navbar implementation (80%, needing backend functionality now...)
- README file updates

### CODE REVIEW

**Comment by JDoan03** [comment]
[Navbar updates](https://github.com/hashemJaber/SacOverflow/pull/5/commits/055f4fa760e75d90536559febe42860d9e38b704) looks good

**Comment by JDoan03** [comment]
[SO-53 Refactoring](https://github.com/hashemJaber/SacOverflow/pull/5/commits/88240e9f568b53aa44fa83e4c576dec69f6acc4e) Read Me is easy to follow along

**Comment by JDoan03** [comment]
Other changes look good. When shrinking to the mobile page size, the left side navbar doesn't go away and overlaps other items. May want to fix before commiting

File: src/components/Dashboard/project_activity/project_activity.tsx [hunk 1]
```diff
@@ -3,94 +3,283 @@
...
-            className={`cursor-pointer py-1 px-3 rounded-full ${currentMonth === month ? 'bg-blue-500 text-white' : 'hover:bg-gray-100'}`}
-          >
-            {month}
-          </span>
-        ))}
-      </div>
-    </div>
-  );
+	return (
+		<div className="max-w-4xl mx-auto p-6 bg-white shadow-lg rounded-lg">
```
Comment (jvniorrr) [inline src/components/Dashboard/project_activity/project_activity.tsx:211]: Add the css properties to the appropriate files for better structure and if need debugging later on... @KiranKaur3
```suggestion
		<div className="project-activity-container">
```

File: src/components/Dashboard/project_activity/project_activity.tsx [hunk 2]
```diff
@@ -3,94 +3,283 @@
 import React, { useState } from 'react';
 
 const ProjectActivity = () => {
```
Comment (jvniorrr) [inline src/components/Dashboard/project_activity/project_activity.tsx:5]: Case 0: If no data is present...
Case 1: Data is present

File: src/components/Dashboard/revenue_linechart/linechart.tsx [hunk 3]
```diff
@@ -6,189 +6,189 @@ import * as am5xy from '@amcharts/amcharts5/xy';
...
-    );
+	let { id, className } = props;
+	const chartRef = useRef(null);
+
+	useLayoutEffect(() => {
+		let root = am5.Root.new(`${id}`);
+
+		root.setThemes([
+			am5themes_Animated.new(root),
+			// am5themes_Responsive.new(root, {
```
Comment (jvniorrr) [inline src/components/Dashboard/revenue_linechart/linechart.tsx:17]: Remove all un needed code @KiranKaur3

File: src/components/Dashboard/revenue_linechart/linechart.tsx [hunk 4]
```diff
@@ -6,189 +6,189 @@ import * as am5xy from '@amcharts/amcharts5/xy';
...
+					}),
+				});
+			});
+
+			chartRef.current = chart;
+		}
+
+		// Set data
+		let chartData = [
+			{
```
Comment (jvniorrr) [inline src/components/Dashboard/revenue_linechart/linechart.tsx:143]: Create an interface object for the data fields to come in. @KiranKaur3

File: src/components/projects/AddCard.tsx [hunk 5]
```diff
@@ -1,26 +1,29 @@
...
-              Add New Project Here...
-            </div>
-          </div>
-        </div>
-      </Link>
-    </div>
-  );
+	return (
+		<div className="bg-gray-100 hover:bg-gray-300 rounded-md shadow-lg p-3 m-2">
+			<Link
```
Comment (jvniorrr) [inline src/components/projects/AddCard.tsx:9]: We will have to tweak this once we have the modal for creating a new project. This card will act as an overlay over the page. So may need to add a prop that is a function handling interaction... TBD @JDoan03

**Review by hashemJaber — APPROVED** [review]
Good work,

**Review by hashemJaber — APPROVED** [review]
Great Job Fernando

**Review by hashemJaber — APPROVED** [review]
Good Job

**Pull request merged**